pkg/mutation/assign_mutator.go

```diff
@@ -0,0 +1,115 @@
+package mutation
+
+import (
+	"github.com/google/go-cmp/cmp"
+	mutationsv1 "github.com/open-policy-agent/gatekeeper/apis/mutations/v1alpha1"
+	"github.com/open-policy-agent/gatekeeper/pkg/mutation/path"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type assignMutator struct {
+	id       ID
+	assign   *mutationsv1.Assign
+	path     []path.Entry
+	bindings []SchemaBinding
+}
+
+// assignMutator implements mutatorWithSchema
+var _ MutatorWithSchema = &assignMutator{}
+
+func (m *assignMutator) Matches(obj metav1.Object, ns *corev1.Namespace) bool {
+	// TODO implement using matches function
+	return false
+}
+
+func (m *assignMutator) Mutate(obj *unstructured.Unstructured) error {
+	// TODO implement
+	return nil
+}
+func (m *assignMutator) ID() ID {
+	return m.id
+}
+
+func (m *assignMutator) SchemaBindings() []SchemaBinding {
+	return m.bindings
+}
+
+func (m *assignMutator) HasDiff(mutator Mutator) bool {
+	toCheck, ok := mutator.(*assignMutator)
+	if !ok { // different types, must be different
+		return true
+	}
+
+	// these may be redundant as generated by
+	// assign
+	if !cmp.Equal(toCheck.id, m.id) {
+		return true
+	}
+	if !cmp.Equal(toCheck.path, m.path) {
+		return true
+	}
+	if !cmp.Equal(toCheck.bindings, m.bindings) {
+		return true
+	}
+
+	// any difference in spec may be enough
+	if !cmp.Equal(toCheck.assign.Spec, m.assign.Spec) {
+		return true
+	}
+
+	return false
+}
+
+func (m *assignMutator) Path() []path.Entry {
+	return m.path
+}
+
+func (m *assignMutator) DeepCopy() Mutator {
+	res := &assignMutator{
+		id:       m.id,
+		assign:   m.assign.DeepCopy(),
+		path:     make([]path.Entry, len(m.path)),
+		bindings: make([]SchemaBinding, 0),
+	}
+	copy(res.path, m.path)
+	copy(res.bindings, m.bindings)
+	return res
+}
+
+func mutatorForAssign(assign *mutationsv1.Assign) (MutatorWithSchema, error) {
```

Comment:
We should return type `assignMutator` rather than the `MutatorWithSchema` interface, per golang conventions.